### DISCUSSION TITLE
Storage Questions

### DISCUSSION THREAD

**Discussion by CryptoDevving** [0xPolygon/polygon-edge#1092] [General]
Is there a way to optimize storage or prune unused blocks from storage? I am curious about some work arounds. It seems that the blockchain becomes very large after being up for months. Is there anyway to address the storage concerns?

**Comment by ivanbozic21** [comment]
Hi @CryptoDevving 

First of all, can you let us know which directory is using most of the disk space?

**Reply by CryptoDevving** [reply]
All bootnode directories. I assume it is normal for it to get to around 4GB or so. But if that is over a few months and we expand that to years it will be terabytes. Which gets expensive for storage.

**Comment by dankostiuk** [comment]
Hi @ivanbozic21 ,

I'm also curious if your team has any future plans relating to pruning or optimizing storage? It does seem like it'll be an issue in a year or so for those of us that have been running polygon-edge since the early days..

**Comment by Solidityghost998** [comment]
Hi @ivanbozic21  
i am running a ibft server with polygon edge inside the docker , and keep running into below mentioned error can you please help with storage needs. 
polygon.server.ibft: cannot write block: err="write /test-chain/test-chain-data-2/blockchain/000003.log: no space left on device.
